docs/design-proposals/2022-04 TraceQL Concepts.md

```diff
@@ -0,0 +1,203 @@
+
+If the trace does contain spans with an attribute `http.status` with the value `200` then only those spans will be returned. The trace is reduced to only the set of spans that match the condition inside the `{}`. The resultset will contain only this subset of spans matching the condition.
+
+### Intrinsic fields
+
+Every span has certain intrinsic fields that can be referenced when selecting spans
+
+| field name | description                                 |
+| ---------- | ------------------------------------------- |
+| `start`    | start time of the span                      |
+| `end`      | end time of the span                        |
+| `duration` | end - start time of the span                |
+| `name`     | operation or span name                      |
+| `status`   | status values are "error", "ok", or "unset" |
+| `parent`   | the parent of this span                     |
+
+**Examples**
+Find traces that contain spans whose duration is greater than 2 seconds:  
+`{ duration > 2s }`
+
+Find traces that contain a span named "HTTP POST":  
+`{ name = "HTTP POST" }`
+
+### Attribute fields
+
+We can refer to dynamic attributes (also known as tags) on the span or the span's resource.
+
+**Examples**
+
+Find traces with the GET http method:  
+`{ http.method = "GET" }`
+
+Find traces that passed through the prod namespace:  
+`{ namespace = "prod" }`
+
+Find traces that cross service boundaries:  
+`{ parent.service.name != service.name }`
+
+### Field expressions
+
+Fields can also be combined in various expected ways.
+
+**Examples**
+
+Find traces where the difference of the duration of a set of parent/child spans exceeds 500ms:  
+`{ parent.duration - duration > 500ms }`
+
+Find traces with "success" http status codes:  
+`{ http.status >= 200 && http.status < 300 }`
+
+Note that the second expression requires both conditions to be true on the same span. The entire expression inside of `{}` must be evaluated as true on a single span for it to be included in the resultset.
+
+## Combining Spansets
+
+Logical operators combine sets of spans. For example, if we wanted to find a trace that went through two specific regions:  
+`{ region = "eu-west-0" } && { region = "eu-west-1" } `
+
+Note the difference between the above and the following:  
+`{ region = "eu-west-0"  && region = "eu-west-1" } `
+
+The second expression will return no traces because it's impossible for both conditions to be simultaneously true on the same span.
+
+### Structural operators
+
+Structural operators evaluate traces based on the span relationships within the span tree. 
+
+`{ } >> { }`  
+  This is the descendant operator. The spans returned from this operator will match the right hand side conditions while also being descendants of spans that match the left hand side conditions.
+
+`{ } > { }`  
```

Comment:
While `>` for direct parent-child relation looks intuitive when thinking of spans, I can imagine that overloading the `>` character will confuse users with less familiarity with the trace model.